src/Azure/Orleans.Persistence.AzureStorage/Providers/Storage/AzureBlobStorage.cs

```diff
@@ -173,35 +179,38 @@ public async Task ClearStateAsync(string grainType, GrainReference grainId, IGra
             }
         }
 
-        private async Task WriteStateAndCreateContainerIfNotExists(string grainType, GrainReference grainId, IGrainState grainState, byte[] contents, CloudBlockBlob blob)
+        private async Task WriteStateAndCreateContainerIfNotExists(string grainType, GrainReference grainId, IGrainState grainState, byte[] contents, string mimeType, BlobClient blob)
         {
             try
             {
-                await DoOptimisticUpdate(() => blob.UploadFromByteArrayAsync(contents, 0, contents.Length, AccessCondition.GenerateIfMatchCondition(grainState.ETag), null, null),
+                using var stream = new MemoryStream(contents);
```

Comment:
`MemoryStream` does not need to be disposed, so this can be inlined: `... => blob.UploadAsync(new MemoryStream(contents), ...`

Reply:
Disagree :) It implements `IDisposable` so it should be disposed. If you ever add analyzers, it will be flagged with CA2000.